### PR TITLE
DXP-306 DX Improvements

### PR DESCRIPTION
**Pull request by jmontoyaa**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -203,13 +204,22 @@ export async function getApp(
 
   app.setErrorHandler(errorHandler)
 
+  const dependencies: DependencyOverrides = configOverrides
```
Comment (kibertoad) [inline src/app.ts]: wait, how is this different from https://github.com/lokalise/node-service-template/pull/671/files?

Comment (jmontoyaa) [reply src/app.ts]: It's the same thing. My branch was outdated, after merge I got the new changes from 671

File: src/modules/users/consumers/PermissionsConsumer.spec.ts [hunk 2]
```diff
@@ -14,11 +14,12 @@ import { getApp } from '../../../app.js'
 import { SINGLETON_CONFIG } from '../../../infrastructure/parentDiConfig.js'
 import type { PermissionsService } from '../services/PermissionsService.js'
 
+import { generateMonotonicUuid } from '@lokalise/id-utils'
```
Comment (dariacm) [inline src/modules/users/consumers/PermissionsConsumer.spec.ts:17]: Is `@lokalise/id-utils` an existing dependency? I am not seeing it in the `package.json`

Comment (jmontoyaa) [reply src/modules/users/consumers/PermissionsConsumer.spec.ts:17]: Good catch, indeed. Added: https://github.com/lokalise/node-service-template/pull/683/commits/f722889acef2133f3e91e515301b1d8c0f5f3d7e

**Review by dariacm — APPROVED** [review]

File: src/modules/users/consumers/PermissionsConsumer.spec.ts [hunk 3]
```diff
@@ -14,11 +14,12 @@ import { getApp } from '../../../app.js'
...
 import type { PublisherManager } from '../../../infrastructure/commonDiConfig.js'
 import { buildQueueMessage } from '../../../utils/queueUtils.js'
 import { PermissionConsumer } from './PermissionConsumer.js'
 
-const userIds = ['100', '200', '300']
+const userIds = [generateMonotonicUuid(), generateMonotonicUuid(), generateMonotonicUuid()]
```
Comment (kibertoad) [inline src/modules/users/consumers/PermissionsConsumer.spec.ts:22]: maybe let's use generateUuid7() instead?

Comment (jmontoyaa) [reply src/modules/users/consumers/PermissionsConsumer.spec.ts]: Updated: https://github.com/lokalise/node-service-template/pull/683/commits/7ed11e00561288d89dd2b742e09d0c4950bbecc3

**Review by kibertoad — APPROVED** [review]

**Pull request merged**